plugins/renepay/mcf.c

```diff
@@ -483,11 +483,18 @@ static bool linearize_channel(const struct pay_parameters *params,
 	a = MAX(a,0);
 	b = MAX(a+1,b);
 
+	/* An extra bound on capacity, here we use it to reduce the flow such
+	 * that it does not exceed htlcmax. */
+	s64 cap_on_capacity =
+	 channel_htlc_max(c, dir).millisatoshis/1000; /* Raw: linearize_channel */
+
 	capacity[0]=a;
 	cost[0]=0;
 	for(size_t i=1;i<CHANNEL_PARTS;++i)
 	{
-		capacity[i] = params->cap_fraction[i]*(b-a);
+		capacity[i] = MIN(params->cap_fraction[i]*(b-a), cap_on_capacity);
+		cap_on_capacity -= capacity[i];
```

Comment:
In this loop we are splitting a channel into 4 arcs with increasing cost slopes, aka linearization of the cost function. At this point we are limiting the maximum flow that can be allocated in this channel such that it does not exceed `htlc_max`, that is the sum of all arcs capacity does not exceed `htlc_max`.

We could instead constrain the individual arc capacities to not exceed `htlc_max`, that would imply that we are allowing up to 4 times `htlc_max` to go through this channel, but in separate hltcs. And because each successive arc has a higher slope than the previous one, we would have a higher cost associated to the second HTLC with respect to the first, and so on, thus discouraging a second or third or fourth HTLC allocated in the same channel.
```suggestion
```

Reply:
Agreed.  htlc_max is respected as a courtesy, not a necessity, especially if there are few other options.

Reply:
I am reconsidering on this. 
According to BOLT02: the total value of the HTLCs should not exceed `max_htlc_value_in_flight`,
but `max_htlc_value_in_flight` for remote nodes it not visible, the closest proxy we have for it is `htlc_max` which satisfies `htlc_max<max_htlc_value_in_flight`. Therefore I think it is better to play nice here and assume that we cannot send more than `htlc_max` per payment attempt.